GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs

```diff
@@ -28,69 +35,113 @@ private void cbIncludeSubmodules_CheckedChanged(object sender, EventArgs e)
 
         private void FetchData()
         {
-            Loading.Visible = true;
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            _backgroundLoaderTokenSource.Cancel();
```

Comment:
`+ _backgroundLoaderTokenSource.Dispose();`

Reply:
Thanks, done.

Looking through the code there are many places where `CancellationTokenSource` isn't disposed. There are also places where tokens are disposed but not cancelled (e.g. `AsyncLoader`).

The dispose actually does deterministically clean up `ManualResetEvent m_kernelEvent` which is likely a good thing.

All the more reason to pull the pattern out into something simpler, like `CancellationTokenSequence` from #4735.

Reply:
:bulb: The last time I checked, `Dispose` is only actually needed if the source was created with `CreateLinkedTokenSource`. This isn't a restriction you should be assuming when you write new code but it helps set the work priority for cleaning up existing cases.